frontend/src/__tests__/unit/utils/string.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { replaceNonNumericPartWithString, replaceNumericPartWithString } from '~/utilities/string';
+
+describe('replaceNumericPartWithString', () => {
+  it('should replace the numeric part of a string with a number', () => {
+    expect(replaceNumericPartWithString('abc123xyz', 456)).toBe('abc456xyz');
+  });
+
+  it('should handle empty input string', () => {
+    expect(replaceNumericPartWithString('', 789)).toBe('789');
+  });
+
+  it('should handle input string without numeric part', () => {
+    expect(replaceNumericPartWithString('abcdef', 123)).toBe('123abcdef');
+  });
+
+  it('should handle numeric part at the beginning of the string', () => {
+    expect(replaceNumericPartWithString('123xyz', 789)).toBe('789xyz');
+  });
+
+  it('should handle numeric part at the end of the string', () => {
+    expect(replaceNumericPartWithString('abc456', 123)).toBe('abc123');
+  });
+
+  it('should handle zero numeric replacement', () => {
+    expect(replaceNumericPartWithString('abc0xyz', 123)).toBe('abc123xyz');
```

Comment:
What does this test 🤔 You have `abc123xyz` test... a single to multiple replacement? The test name does not state that... why is `0` special?

Reply:
There's nothing particularly unique about `0`; I consider it just one of the test cases. I'll remove it since it's similar to the other test cases